### PR TITLE
[lr] Add Linear Learning Rate Scheduler

### PR DESCRIPTION
**Pull request by lhw414**
This PR adds the [LinearLearningRateScheduler](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LinearLR.html) class, which adjusts the learning rate using a Linear formula.

The implementation includes setting the max/min learning rates, decay steps, and proper error handling for missing properties.

And add unittest cases(property setting test, finalize test, getLearningRate test).

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/optimizers/lr_scheduler_linear.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+public:
+  /**
+   * @brief Construct a new Linear Learning Rate Scheduler object
+   */
+  LinearLearningRateScheduler();
+
+  /**
+   * @copydoc LearningRateScheduler::getLearningRate(size_t iteration) const
+   */
+  virtual double getLearningRate(size_t iteration) override;
```
Comment (baek2sm) [inline nntrainer/optimizers/lr_scheduler_linear.h:40]: how about remove "virtual" keyword on this function?

as far as I know, many C++ style guides recommend specifying either 'override' or 'final', but not using 'virtual'.

File: nntrainer/optimizers/lr_scheduler_linear.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  /**
+   * @copydoc LearningRateScheduler::finalize()
+   */
+  virtual void finalize() override;
```
Comment (baek2sm) [inline nntrainer/optimizers/lr_scheduler_linear.h:45]: ditto

**Review by baek2sm — APPROVED** [review]
LGTM!

File: nntrainer/optimizers/lr_scheduler.h [hunk 3]
```diff
@@ -92,6 +93,11 @@ class LearningRateScheduler : public ml::train::LearningRateScheduler {
    * - min_learning_rate : float
    * - decay_steps : float
    *
+   * Linear Learning rate scheduler
+   * - max_learning_rate : float
+   * - min_learning_rate : float
+   * - decay_steps : float
```
Comment (djeong20) [inline nntrainer/optimizers/lr_scheduler.h:99]: decay_steps is stated as a float. shouldn't it be an integer?

File: nntrainer/optimizers/lr_scheduler_linear.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+void LinearLearningRateScheduler::finalize() {
+  NNTR_THROW_IF(std::get<props::MaxLearningRate>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Max Learning Rate is not set";
+  NNTR_THROW_IF(std::get<props::MinLearningRate>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Min Learning Rate is not set";
+  NNTR_THROW_IF(std::get<props::DecaySteps>(lr_props).empty(),
+                std::invalid_argument)
+    << "[LinearLearningRateScheduler] Decay Steps is not set";
```
Comment (djeong20) [inline nntrainer/optimizers/lr_scheduler_linear.cpp:37]: it would be nice if you also check that decay_steps is positive.

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp [hunk 5]
```diff
@@ -478,6 +479,72 @@ TEST(lr_cosine, getlearningrate_01_n) {
...
+
+TEST(lr_linear, prop_02_n) {
+  auto lr = createLRS("linear");
+  EXPECT_THROW(lr->setProperty({"learning_rate:0.1"}), std::invalid_argument);
+}
+
+TEST(lr_linear, prop_03_n) {
+  auto lr = createLRS("linear");
+  EXPECT_NO_THROW(lr->setProperty({"max_learning_rate=1.0"}));
+}
```
Comment (djeong20) [inline test/unittest/unittest_nntrainer_lr_scheduler.cpp:496]: `_n` stands for negative here. please check test cases and change it to `_p` if it's a positive test case!

File: nntrainer/optimizers/lr_scheduler_linear.h (hunk 1 above)
Comment (lhw414) [reply nntrainer/optimizers/lr_scheduler_linear.h]: Removed 'virtual' keyword

File: nntrainer/optimizers/lr_scheduler_linear.h (hunk 2 above)
Comment (lhw414) [reply nntrainer/optimizers/lr_scheduler_linear.h]: Removed 'virtual' keyword

File: nntrainer/optimizers/lr_scheduler.h (hunk 3 above)
Comment (lhw414) [reply nntrainer/optimizers/lr_scheduler.h]: changed it 'positive integer'

File: nntrainer/optimizers/lr_scheduler_linear.cpp (hunk 4 above)
Comment (lhw414) [reply nntrainer/optimizers/lr_scheduler_linear.cpp:37]: Added it

File: test/unittest/unittest_nntrainer_lr_scheduler.cpp (hunk 5 above)
Comment (lhw414) [reply test/unittest/unittest_nntrainer_lr_scheduler.cpp:496]: Fixed it. Thank you!

**Review by djeong20 — APPROVED** [review]
LGTM! great job :)

**Review by jijoongmoon — APPROVED** [review]
LGTM
